pypeit/coadd1d.py

```diff
@@ -103,68 +204,46 @@ def load(self):
 
     def save(self, coaddfile, telluric=None, obj_model=None, overwrite=True):
         """
-        Routine to save 1d coadds to a fits file. This replaces save.save_coadd1d_to_fits
+        Generate a :class:`OneSpec` object and write it to disk.
 
         Args:
             coaddfile (str):
-               File to outuput coadded spectrum to.
-            telluric (str):
-               This is vestigial and should probably be removed.
+               File to output coadded spectrum to.
+            telluric (`numpy.ndarray`_):
             obj_model (str):
-               This is vestigial and should probably be removed
             overwrite (bool):
                Overwrite existing file?
-
         """
-
         self.coaddfile = coaddfile
-        ex_value = self.par['ex_value']
-        # Estimate sigma from ivar
-        sig = np.sqrt(utils.inverse(self.ivar_coadd))
-        if (os.path.exists(self.coaddfile)) and (np.invert(overwrite)):
-            hdulist = fits.open(self.coaddfile)
-            msgs.info("Reading primary HDU from existing file: {:s}".format(self.coaddfile))
-        else:
-            msgs.info("Creating a new primary HDU.")
-            prihdu = fits.PrimaryHDU()
-            if self.header is None:
-                msgs.warn('The primary header is none')
-            else:
-                prihdu.header = self.header
-            hdulist = fits.HDUList([prihdu])
-
         wave_mask = self.wave_coadd > 1.0
```

Comment:
Since we want to remove pixels with wavelength equals zero in the coadd1d, we made a separate mask for them. This line works for both multisite and Echelle

Reply:
Yeah look further down when we are writing these out to fits. It seems like it is writing out arrays that do not have the right shape, which is what is confusing me about it.

Reply:
out arrays have the same shape to me.